vyked/decorators/tcp.py

```diff
@@ -68,6 +68,28 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _trace_id_getter(task):
+    trace_id = getattr(task, 'trace_id', None)
+    if trace_id:
+        return trace_id
+    else:
+        try:
```

Comment:
@amanwriter just so that we are on the same page. The purpose of trace_id is that it is initially assigned the first time a request comes into a _vyked service cloud_ and the same one is used for all calls that occur between services in order to fulfil the original request so that we can trace what happened for any specific request.

The way I was originally thinking about it the trace_id only needs to be checked when a request is received and if it isn't there it just gets added when the response is sent. This can happen purely at the packet level and hence all of this should not be necessary.

PS: Also should be done for pub subs also.